### PR TITLE
review register until CTRL8_XL

### PR DESCRIPTION
**Pull request by deddy11**
Hello, please be kindly to review and accept my pull request. I did some formatting changes, I don't know they are typos or your coding style. So please forgive me if it is indeed a coding style.

### CODE REVIEW

File: src/ISM330DLCSensor.cpp [hunk 1]
```diff
@@ -418,7 +418,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Get_G_Sensitivity(float *pfData)
     /* Store the sensitivity based on actual full scale. */
     switch( fullScale )
     {
-      case ISM330DLC_ACC_GYRO_FS_G_245dps:
+      case ISM330DLC_ACC_GYRO_FS_G_250dps:
         *pfData = ( float )ISM330DLC_GYRO_SENSITIVITY_FOR_FS_245DPS;
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: I would change also the ISM330DLC_GYRO_SENSITIVITY_FOR_FS_245DPS define in ISM330DLC_GYRO_SENSITIVITY_FOR_FS_250DPS

File: src/ISM330DLCSensor.cpp [hunk 2]
```diff
@@ -504,7 +504,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Get_X_ODR(float* odr)
     case ISM330DLC_ACC_GYRO_ODR_XL_POWER_DOWN:
       *odr = 0.0f;
       break;
-    case ISM330DLC_ACC_GYRO_ODR_XL_13Hz:
+    case ISM330DLC_ACC_GYRO_ODR_XL_12Hz5:
       *odr = 13.0f;
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need to replace "*odr = 13.0f;" with "*odr = 12.5f;"

File: src/ISM330DLCSensor.cpp [hunk 3]
```diff
@@ -561,7 +561,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Get_G_ODR(float* odr)
     case ISM330DLC_ACC_GYRO_ODR_G_POWER_DOWN:
       *odr = 0.0f;
       break;
-    case ISM330DLC_ACC_GYRO_ODR_G_13Hz:
+    case ISM330DLC_ACC_GYRO_ODR_G_12Hz5:
       *odr = 13.0f;
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need to replace "*odr = 13.0f;" with "*odr = 12.5f;"

File: src/ISM330DLCSensor.cpp [hunk 4]
```diff
@@ -633,7 +633,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Set_X_ODR_When_Enabled(float odr)
 {
   ISM330DLC_ACC_GYRO_ODR_XL_t new_odr;
 
-  new_odr = ( odr <=   13.0f ) ? ISM330DLC_ACC_GYRO_ODR_XL_13Hz
+  new_odr = ( odr <=   13.0f ) ? ISM330DLC_ACC_GYRO_ODR_XL_12Hz5
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need also to modify the condition in this way: "new_odr = ( odr <= 12.5f ) ? ISM330DLC_ACC_GYRO_ODR_XL_12Hz5"

File: src/ISM330DLCSensor.cpp [hunk 5]
```diff
@@ -707,7 +707,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Set_G_ODR_When_Enabled(float odr)
 {
   ISM330DLC_ACC_GYRO_ODR_G_t new_odr;
 
-  new_odr = ( odr <=  13.0f )  ? ISM330DLC_ACC_GYRO_ODR_G_13Hz
+  new_odr = ( odr <=  13.0f )  ? ISM330DLC_ACC_GYRO_ODR_G_12Hz5
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need also to modify the condition in this way: "new_odr = ( odr <= 12.5f ) ? ISM330DLC_ACC_GYRO_ODR_G_12Hz5"

File: src/ISM330DLCSensor.cpp [hunk 6]
```diff
@@ -811,7 +811,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Get_G_FS(float* fullScale)
   {
     switch( fs_low_level )
     {
-      case ISM330DLC_ACC_GYRO_FS_G_245dps:
+      case ISM330DLC_ACC_GYRO_FS_G_250dps:
         *fullScale = 245.0f;
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need to replace "*fullScale = 245.0f;" with "*fullScale = 250.0f;"

File: src/ISM330DLCSensor.cpp [hunk 7]
```diff
@@ -872,7 +872,7 @@ ISM330DLCStatusTypeDef ISM330DLCSensor::Set_G_FS(float fullScale)
   }
   else
   {
-    new_fs = ( fullScale <=  245.0f ) ? ISM330DLC_ACC_GYRO_FS_G_245dps
+    new_fs = ( fullScale <=  245.0f ) ? ISM330DLC_ACC_GYRO_FS_G_250dps
```
Comment (cparata) [inline src/ISM330DLCSensor.cpp]: we need also to modify the condition in this way: "new_fs = ( fullScale <= 250.0f ) ? ISM330DLC_ACC_GYRO_FS_G_250dps"

**Comment by cparata** [comment]
Hello @deddy11 ,
thanks a lot for your PR! It seems to me very good. I would just ask you few changes. Could you update the PR, so I can merge it in the main branch?
Thanks in advance.
Carlo

**Comment by deddy11** [comment]
Hi @cparata ,
thanks for your reviews. I have updated the PR.

**Comment by cparata** [comment]
If you agree, I will release a new version of the library with all these changes

**Comment by deddy11** [comment]
Please wait for my next PR XD. I am still reviewing for the rest register

**Pull request merged**